pkg/controller/argocd/role.go

```diff
@@ -85,6 +86,21 @@ func (r *ReconcileArgoCD) reconcileRoles(cr *argoprojv1a1.ArgoCD) (role *v1.Role
 
 // reconcileRole
 func (r *ReconcileArgoCD) reconcileRole(name string, policyRules []v1.PolicyRule, cr *argoprojv1a1.ArgoCD) (*v1.Role, error) {
+	if name == applicationController {
+		if err := applyReconcilerHook(cr, &policyRules, "policyRulesForApplicationController"); err != nil {
+			return nil, err
+		}
+		if !reflect.DeepEqual(policyRules, policyRuleForApplicationController()) {
+			admin := &v1.ClusterRole{}
+			if err := r.client.Get(context.TODO(), types.NamespacedName{Name: "admin"}, admin); err != nil {
```

Comment:
clusterrole admin is openshift specific.  It should be handled in the openshift hook.

Reply:
moved this to openshift hooks!!